src/v/redpanda/debug_bundle.cc

```diff
@@ -11,16 +11,123 @@
+    if (!host_env) {
+        vlog(
+          bundle_log.warn,
+          "Failed to get 'PATH' environmental variable, the debug bundle may "
+          "be incomplete due to missing dependencies");
+    } else {
+        host_path = fmt::format("PATH={}", host_env);
+    }
+
+    auto filename = fmt::format(
+      "{}/{}", _write_dir.string(), _in_progress_filename);
+    std::vector<ss::sstring> rpk_argv{
+      _rpk_cmd.string(), "debug", "bundle", "--output", filename};
+
+    ssx::background
+      = ss::experimental::spawn_process(
+          _rpk_cmd, {.argv = std::move(rpk_argv), .env = {host_path}})
+          .then([this](auto process) {
+              auto stdout = process.stdout();
+              return ss::do_with(
+                std::move(process),
+                std::move(stdout),
+                [this](auto& p, auto& stdout) {
+                    gate_guard guard{_rpk_gate};
+                    return stdout.consume(rpk_consumer())
+                      .finally([this, &p, guard{std::move(guard)}]() mutable {
+                          return p.wait()
+                            .then([this](ss::experimental::process::wait_status
+                                           wstatus) {
+                                auto* exit_status = std::get_if<
+                                  ss::experimental::process::wait_exited>(
+                                  &wstatus);
+                                if (exit_status == nullptr) {
+                                    vlog(
+                                      bundle_log.error,
+                                      "Failed to run RPK, exit status is "
+                                      "undefined");
+                                }
+
+                                if (exit_status->exit_code != 0) {
+                                    vlog(
+                                      bundle_log.error,
+                                      "Failed to run RPK, exit code {}",
+                                      exit_status->exit_code);
+                                } else {
+                                    vlog(
+                                      bundle_log.debug,
+                                      "RPK successfully created debug bundle "
+                                      "{}",
+                                      _in_progress_filename);
+                                }
+                            })
+                            .finally([&p, guard{std::move(guard)}]() mutable {
+                                // Make sure the process dies, first gracefully
+                                // and then forcefully.
+                                // Please note: Seastar reports an ignored
+                                // exceptional future when SIGTERM or SIGKILL is
+                                // called on an already dead process.
+                                p.terminate();
+                                p.kill();
```

Comment:
Ignoring an exceptional future seems dirty to me. I could put a `handle_exception` here to silence it but that's even more filthy.